release-notes/4.0.0.md

```diff
@@ -0,0 +1,55 @@
+ - the `Rule` keyword acts as a grouping of scenarios inside a feature
+ - the `Scenario Outline` keyword is not needed anymore when using `Examples` with a `Scenario``
+ - the `Scenario` keyword now has synonym: `Example`
+
+If you are familiar with [example mapping](https://cucumber.io/docs/bdd/example-mapping/), you will see how easy it is to translate your cards from an example mapping session into a `feature` file.
+
+### support of the cucumber-messages protocol
+
+[`cucumber-messages`](https://github.com/cucumber/cucumber/tree/master/messages#cucumber-messages) is the new format that is used by Cucumber to represent the various information emitted during an execution.
+
+This library satisfies multiple interests:
+ - same output for all `cucumber` and related implementation (cucumber-ruby, cucumber-jvm, cucumber-jss, SpecFlow ...)
+ - it relies on [`protobuf`](https://developers.google.com/protocol-buffers/), so it is fairly easy to generate a consumer in any language supported by `protobuf`.
+  - we have published libraries to consume this format for [`ruby`](https://rubygems.org/gems/cucumber-messages), [`Java`](https://mvnrepository.com/artifact/io.cucumber/messages), [`JavaScript](https://www.npmjs.com/package/@cucumber/messages) and [`C#`](https://www.nuget.org/packages/Cucumber.Messages/)
+ - it is easier to generate tools consuming this output
+
+To generate the new messages, you can use the `message` reporter.
+
+Example: `bundle exec cucumber --format message`
+
+### sending results to a server
+
+You now have the option to send the results of the execution directly to a web server. You can achieve this by specifying an `http` adress to the `--out` parameter.
+
+Example: `bundle exec cucumber --format messages --out "http://example.com/report-service"`
+
+### HTML reporter
+
+We've added a brand new HTML reporter in this release. Everything is embedded in a single file (no CSS or JS file), so you can easily share the generated page.
+
+Example: `bundle exec cucumber --format html --out report.html`
+
+## Deprecated
+
+### JSON formatter
+
+With the introduction of the `message` reporter, we have decided to deprecate the JSON reporter.
+We have released an executable tool ([`json-formatter`](https://github.com/cucumber/cucumber/tree/master/json-formatter#cucumber-json-formatter)) to produce old JSON from the new `message` format.
+
+### `puts`
+
+When using `puts` in your step definitions, the data you output are caught by `cucumber` and then processed by the formatter (for example, when using the `json` formatter, the data provided to `puts` will appear in the `output` field of the step).
+This solution is now deprecated in favor of the `log` method. In `cucumber-ruby 5`, calls to `puts` will not be caught by the formatter and the default Ruby behavior will be triggered.
```

Comment:
Woop!

Reply:
"In version 3.x, when using `puts` in your step definitions or hooks"
"is caught by"